src/Tools/AI Test Toolkit/src/TestSuite/AITALTestSuiteMgt.Codeunit.al

```diff
@@ -0,0 +1,202 @@
+    begin
+        this.RemoveTestMethods(AITTestMethodLine, 0, '');
+    end;
+
+    internal procedure RemoveTestMethods(var AITTestMethodLine: Record "AIT Test Method Line"; CodeunitID: Integer; DataInputName: Text[250])
+    var
+        TestMethodLine: Record "Test Method Line";
+    begin
+        TestMethodLine.SetRange("Test Suite", AITTestMethodLine."AL Test Suite");
+        if CodeunitID > 1 then
+            TestMethodLine.SetRange("Test Codeunit", CodeunitID);
+
+        if DataInputName <> '' then
+            TestMethodLine.SetRange("Data Input", DataInputName);
+
+        TestMethodLine.ReadIsolation := TestMethodLine.ReadIsolation::ReadUncommitted;
+        if TestMethodLine.IsEmpty() then
+            exit;
+
+        TestMethodLine.DeleteAll();
+        this.RemoveEmptyCodeunitTestLines(this.GetOrCreateALTestSuite(AITTestMethodLine));
+    end;
+
+    internal procedure RemoveEmptyCodeunitTestLines(ALTestSuite: Record "AL Test Suite")
+    var
+        TestMethodLine: Record "Test Method Line";
+        FunctionTestMethodLine: Record "Test Method Line";
+    begin
+        TestMethodLine.SetRange("Test Suite", ALTestSuite.Name);
+        TestMethodLine.SetRange("Line Type", TestMethodLine."Line Type"::Codeunit);
+        TestMethodLine.ReadIsolation := TestMethodLine.ReadIsolation::ReadUncommitted;
+
+        if not TestMethodLine.FindSet() then
+            exit;
+
+        repeat
+            FunctionTestMethodLine.SetRange("Test Suite", ALTestSuite.Name);
+            FunctionTestMethodLine.SetRange("Test Codeunit", TestMethodLine."Test Codeunit");
+            FunctionTestMethodLine.SetRange("Line Type", FunctionTestMethodLine."Line Type"::Function);
+            FunctionTestMethodLine.ReadIsolation := FunctionTestMethodLine.ReadIsolation::ReadUncommitted;
+            if FunctionTestMethodLine.IsEmpty() then
+                TestMethodLine.Delete();
+        until TestMethodLine.Next() = 0;
+    end;
+
+    internal procedure GetOrCreateALTestSuite(var AITTestMethodLine: Record "AIT Test Method Line"): Record "AL Test Suite"
+    var
+        AITTestSuite: Record "AIT Test Suite";
+        ALTestSuite: Record "AL Test Suite";
+    begin
+        if AITTestMethodLine."AL Test Suite" <> '' then begin
+            ALTestSuite.SetFilter(Name, AITTestMethodLine."AL Test Suite");
+            if ALTestSuite.FindFirst() then
+                exit(ALTestSuite);
+        end;
+
+        if AITTestMethodLine."AL Test Suite" = '' then begin
+            AITTestMethodLine."AL Test Suite" := this.GetUniqueAITTestSuiteCode();
+            AITTestMethodLine.Modify();
+        end;
+
+        ALTestSuite.Name := AITTestMethodLine."AL Test Suite";
+        ALTestSuite.Description := CopyStr(AITTestMethodLine.Description, 1, MaxStrLen(ALTestSuite.Description));
+        AITTestSuite.ReadIsolation := IsolationLevel::ReadUncommitted;
+        if AITTestSuite.Get(AITTestMethodLine."Test Suite Code") then
```

Comment:
Add SetLoadFields

Reply:
Not needed for this purpose. The table is small, there is no gain.